torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -309,11 +309,6 @@ class TensorExprFuser {
 
   bool allShapesAreKnown(Node* node) {
     // TODO: Relax the checks to support dynamic shapes
-    for (Value* output : node->outputs()) {
```

Comment:
Just because you know the input shapes doesnt mean you know the output shapes:
```
def index(i: int):
    x = torch.tensor(1, 2, 3, 4)
    out = x[i]
```

Reply:
It does for the subset of ops NNC supports though, at least currently. When we add more complicated ops, we would need to adjust the fuser logic as well, but as it stands now, I think we always can infer output shapes when we know input shapes.